pvlib/pvsystem.py

```diff
@@ -66,6 +69,57 @@ def _combine_localized_attributes(pvsystem=None, location=None, **kwargs):
     return new_kwargs
 
 
+def return_singleton_value(func):
+    """Decorator that returns the value contained when `func`
+    returns a singleton list or the list if it has more than one
+    element.
+    """
+    @functools.wraps(func)
+    def f(*args, **kwargs):
+        return_list = kwargs.pop('return_list', False)
+        x = func(*args, **kwargs)
+        if len(x) == 1 and not return_list:
+            return x[0]
+        return x
+    return f
+
+
+def validate_against_arrays(*list_params):
+    """Decorator that validates the value passed to each parameter in
+    `list_params` against the number of Arrays in the PVSystem.
+
+    If the value passed for each parameter in `list_params` is not a list,
+    then it is transformed into a singleton list before validation. This
+    means existing code that assumes a PVSystem has only one array will
+    continue to function with no changes.
+
+    Implicitly applies the `@singleton_as_scalar` decorator as well.
```

Comment:
Forgot to update docstring after name change.

```suggestion
    Implicitly applies the `@return_singleton_value` decorator as well.
```